### PR TITLE
py-notebook: add 6.5.4

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/jupyter/notebook/tree/v6.5.4

The only dependency that changed was `py-nbclassic` the rest is just reordered.

Regarding `py-setuptools` as run dependency: I am not sure if it is still needed, I don't see it anywhere any don't know where it came from, so I left it.

### CODE REVIEW

**Comment by manuelakuhn** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/py-notebook/package.py [hunk 1]
```diff
@@ -37,42 +38,42 @@ class PyNotebook(PythonPackage):
...
-    depends_on("python@3.5:", type=("build", "run"), when="@6:")
-    depends_on("python@3.6:", type=("build", "run"), when="@6.3:")
     depends_on("python@3.7:", type=("build", "run"), when="@6.4:")
+    depends_on("python@3.6:", type=("build", "run"), when="@6.3:")
     # TODO: replace this after concretizer learns how to concretize separate build deps
     depends_on("py-jupyter-packaging11", when="@6.4.1:", type="build")
     # depends_on('py-jupyter-packaging@0.9:0', when='@6.4.1:', type='build')
-    depends_on("py-setuptools", type=("build", "run"), when="@5:")
+
+    depends_on("py-setuptools", when="@5:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-notebook/package.py]:
```suggestion
    depends_on("py-setuptools", when="@5:", type="build")
```
I don't see any imports either, I would just remove the run-time dep

File: var/spack/repos/builtin/packages/py-notebook/package.py [hunk 2]
```diff
@@ -37,42 +38,42 @@ class PyNotebook(PythonPackage):
...
-    depends_on("py-terminado@0.8.3:", type=("build", "run"), when="@6.1:")
-    depends_on("py-prometheus-client", type=("build", "run"), when="@5.7.0:")
+    depends_on("py-send2trash@1.8:", when="@6.4.10:", type=("build", "run"))
+    depends_on("py-send2trash@1.5:", when="@6.2.0:", type=("build", "run"))
+    depends_on("py-send2trash", when="@6:", type=("build", "run"))
+    depends_on("py-terminado@0.8.3:", when="@6.1:", type=("build", "run"))
+    depends_on("py-terminado@0.8.1:", when="@5.7.0:", type=("build", "run"))
+    depends_on("py-terminado@0.3.3:", when="@:5.7.0", type=("build", "run"))
+    depends_on("py-prometheus-client", when="@5.7.0:", type=("build", "run"))
+    depends_on("py-nbclassic@0.4:", when="@6.5:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-notebook/package.py]: 0.4.7+ now

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-notebook/package.py (hunk 2 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-notebook/package.py]: Ah, thanks. I took the one from `pyproject.toml` and did not crosscheck with the version in `setup.py`. This is fixed now.

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
@spackbot fix style

**Review by adamjstewart — APPROVED** [review]

**Comment by manuelakuhn** [comment]
@spackbot run pipeline

**Pull request merged**